### PR TITLE
Feature/update snowflake args

### PR DESCRIPTION
**Pull request by rwhitten577**
This started as a quest to allow kwargs for the Snowflake connector for a job I'm working on, and here we are with a base pep-249 DB client and a couple concrete implementations 😄  

- BaseConnection class, just a wrapper around PEP-249 spec with addition of `query` and `write_results_to_stream` methods
- Concrete implementations for Snowflake & MySQL, allowing all their native connection args/kwargs
- Started an `omelette/mocks` directory thinking we may want to pull these mocks into different applications, and thus they should live within the `omelette` package

### CODE REVIEW

File: omelette/io.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        if len(rows) % chunk_size == 0:
+            writer.writerows(rows)
+            rows = []
+
+    if len(rows) > 0:
+        writer.writerows(rows)
+
+
+def write_to_stream_json(stream: TextIO, data: Iterable, chunk_size: int):
```
[the remark this answers is not in the document]
Comment (JoshuaSchlichting) [reply omelette/io.py:25]: `write_to_stream_json()` -> `write_to_stream_jsonl()`

File: omelette/db/__init__.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    """Adds missing iterable behavior to pep249.Cursor"""
+
+    def __iter__(self) -> Union[Iterator[Dict], Iterator[Tuple]]:
+        pass
+
+
+class BaseConnection(Connection, ABC):
+    """Wrapper around PEP-249 Connection objects, providing helper methods for common tasks.
+
+    Each inherited child must implement a `connect` method (as directed in PEP-249), that is exported from its module.
```
[the remark this answers is not in the document]
Comment (rwhitten577) [reply omelette/db/__init__.py:28]: Discussed on zoom, this is the PEP-249 spec to provide a constructor called `connect` that returns the Connection object https://www.python.org/dev/peps/pep-0249/#constructors

File: omelette/db/__init__.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    This requires each to be in its own file, such that the following connections do not conflict, e.g.,
+
+    from omelette.db import snowflake, mysql
+
+    mysql_conn = mysql.connect(...)
+    sf_conn = snowflake.connect(...)
```
[the remark this answers is not in the document]
Comment (rwhitten577) [reply omelette/db/__init__.py:34]: Same as above, this follows PEP-249

**Review by JoshuaSchlichting — APPROVED** [review]
LGTM

**Pull request merged**